### PR TITLE
Download engine artifacts in dart code

### PR DESCRIPTION
**Pull request by WonyoungChoi**
context: #77 

- Download tizen engine artifacts by using ArtifactUpdater
- Download gen_snapshot(.exe) from the android's url
- Move the location of tizen artifacts to `flutter/bin/cache/artifacts/engine`
- Remove tizen_artifacts.dart

The artifacts should be uploaded to the github releases in the following form:
- tizen-common.zip
- tizen-x86-debug.zip
- tizen-arm-debug.zip
- tizen-arm-profile.zip
- tizen-arm-release.zip
- tizen-arm64-debug.zip
- tizen-arm64-profile.zip
- tizen-arm64.release.zip

Use following two environment variables:
- `AZURE_BUILD_ID` to download artifacts from the Azure pipeline's artifacts
- `TIZEN_ENGINE_BASE_URL` to override the default url `https://github.com/flutter-tizen/engine/releases`
`AZURE_BUILD_ID` takes precedence over `TIZEN_ENGINE_BASE_URL`

### CODE REVIEW

File: lib/tizen_cache.dart [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+  @override
+  List<String> getLicenseDirs() => const <String>[];
+
+  @override
+  List<String> getPackageDirs() => const <String>[];
+
+  @override
+  Future<void> updateInner(ArtifactUpdater artifactUpdater,
+      FileSystem fileSystem, OperatingSystemUtils operatingSystemUtils) async {
+    const String baseUrl = 'https://github.com/flutter-tizen/engine/releases';
```
[the remark this answers is not in the document]
Comment (WonyoungChoi) [reply lib/tizen_cache.dart:93]: @swift-kim Yes it's possible. but the zip file created by `publish` in the azure pipeline has a root directory always.. like 
```
tizen-arm-release.zip
tizen-arm-release/libflutter_engine.so
```
Can we get rid of it?

[the remark this answers is not in the document]
Comment (swift-kim) [reply lib/tizen_cache.dart:93]: I'll talk to you when I'm back at the office on Thursday.

[the remark this answers is not in the document]
Comment (WonyoungChoi) [reply lib/tizen_cache.dart]: Following two environment variables are added:

- `AZURE_BUILD_ID` to download artifacts from the Azure pipeline's artifacts
- `TIZEN_ENGINE_BASE_URL` to override the default url `https://github.com/flutter-tizen/engine/releases`
`AZURE_BUILD_ID` takes precedence over `TIZEN_ENGINE_BASE_URL`

[the remark this answers is not in the document]
Comment (swift-kim) [reply lib/tizen_cache.dart]: Thank you very much!

File: lib/tizen_cache.dart [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+    }
+  }
+
+  Future<String> _getDownloadUrlFromAzure(String buildId) async {
+    final String azureRestUrl =
+        'https://dev.azure.com/flutter-tizen/flutter-tizen'
+        '/_apis/build/builds/$buildId/artifacts?artifactName=release';
+
+    final http.Response response = await http.get(Uri.parse(azureRestUrl));
+    if (response.statusCode == 200) {
```
Comment (swift-kim) [inline lib/tizen_cache.dart:146]: I searched for the keyword "response.statusCode" in flutter_tools and it seemed more common to use predefined constants:

```suggestion
    if (response.statusCode == HttpStatus.ok) {
```

**Review by swift-kim — APPROVED** [review]
Thank you for your hard work!

**Comment by swift-kim** [comment]
The new release (https://github.com/flutter-tizen/engine/releases/tag/a7ea781) has passed the store verification.

**Pull request merged**